nixio/cmd/upgrade.py

```diff
@@ -0,0 +1,197 @@
+import argparse
+import nixio as nix
+import h5py
+
+
+def get_file_version(fname):
+    with h5py.File(fname, mode="r") as hfile:
+        return tuple(hfile.attrs["version"])
+
+
+def has_valid_file_id(fname):
+    with h5py.File(fname, mode="r") as hfile:
+        fileid = hfile.attrs.get("id")
+        if fileid and nix.util.is_uuid(fileid):
+            return True
+    return False
+
+
+def add_file_id(fname):
+    """
+    Returns a closure that binds the filename. When the return value is
+    called, it adds a UUID to the file header.
+    """
+    def add_id():
+        "Add a UUID to the file header"
+        with h5py.File(fname, mode="a") as hfile:
+            hfile.attrs["id"] = nix.util.create_id()
+    return add_id
+
+
+def update_property_values(fname):
+    """
+    Returns a closure that binds the filename. When the return value is
+    called, it modifies rewrites all the metadata Property objects to the new
```

Comment:
missing "and" like "modifies and rewrites"?

Reply:
OH NO!!

Reply:
welcome